Sources/Views/Text/HTMLView.swift

```diff
@@ -0,0 +1,119 @@
+//
+// This source file is part of the CardinalKit open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+import WebKit
+
+/// An``HTMLView`` allows the display of HTML in a web view including the addition of a header and footer view.
+///
+/// ```
+/// @State var viewState: ViewState = .idle
+///
+/// HTMLView(
+///     asyncHTML: {
+///         // Load your HTML from a remote source or disk storage ...
+///         try? await Task.sleep(for: .seconds(5))
+///         return Data("This is an <strong>HTML example</strong> taking 5 seconds to load.".utf8)
+///     },
+///     state: $viewState,
+///     header: {
+///         Text("Header")
+///     },
+///     footer: {
+///         Text("Footer")
+///     }
+/// )
+/// ```
+private struct WebView: UIViewRepresentable {
+    var htmlContent: String
+
+    func makeUIView(context: Context) -> WKWebView {
+        WKWebView()
+    }
+
+    func updateUIView(_ uiView: WKWebView, context: Context) {
+        uiView.loadHTMLString(htmlContent, baseURL: nil)
+    }
+}
+
+
+public struct HTMLView<Header: View, Footer: View>: View {
+    private let header: Header
+    private let footer: Footer
+    private let asyncHTML: () async -> Data
+
+    @State private var html: Data?
+    @Binding private var state: ViewState
+
+
+    private var htmlString: String {
+        guard let html else {
+            let errorString = String(localized: "HTML_LOADING_ERROR", bundle: .module)
+            return """
+                    <meta name=\"viewport\" content=\"initial-scale=1.0\" />
+                    <p style=\"text-align: center\">\(errorString)</p>
+                    """
+        }
+
+        return String(decoding: html, as: UTF8.self)
+    }
```

Comment:
I would suggest to make `htmlString` optional. The failure path of the guard statement will never be execute as you only access the property in the non-nil path in line 73.
Therefore I would suggest to make the property itself optional and then use an `if let` statement in the view builder to access the property there.

Reply:
Thanks for the suggestion! I have made this change in [edeea29](https://github.com/StanfordBDHG/CardinalKit/pull/39/commits/edeea2905b642bc8d95a6ef7a230490d8c1792d2)